src/storm/storage/BitVector.h

```diff
@@ -564,16 +686,20 @@ class BitVector {
     BitVector(uint_fast64_t bucketCount, uint_fast64_t bitCount);
 
     /*!
-     * Retrieves the index of the next bit that is set to the given value after (and including) the given starting index.
+     * Retrieves the index of the next bit that is set to the given value.
+     * The considered indices are in the range [startIndex, endIndex)
      *
-     * @param value the value of the bit whose index is to be found.
+     * @tparam Value the value of the bit whose index is to be found.
+     * @tparam Backward if true, we search for the bit with closest index to endIndex, otherwise we search for the bit with closest index to startIndex.
      * @param dataPtr A pointer to the first bucket of the data storage.
-     * @param startingIndex The index where to start the search.
-     * @param endIndex The index at which to stop the search.
-     * @return The index of the bit that is set after the given starting index, but before the given end index
-     * in the given bit vector or endIndex in case the end index was reached.
-     */
-    static uint_fast64_t getNextIndexWithValue(bool value, uint64_t const* dataPtr, uint_fast64_t startingIndex, uint_fast64_t endIndex);
+     * @param startingIndex The smallest index that is included in the search range.
+     * @param endIndex The smallest index that is not part of the search range. Should not be larger than the total size of the BitVector.
+     * @return If backwards is false, the index of the found bit is returned (or endIndex in case no bit with Value is present).
+     *         If backwards is true, the index **after** the found bit is returned (or startIndex in case no bit with value Value is present).
+     *         If startIndex >= endIndex, we always return endIndex.
```

Comment:
Sorry to be so nitpicky, but can you update the documentation? It returns endIndex for forwards and startIndex for backwards.

Reply:
Thanks 🥲